src/Haskell/Ide/Engine/Plugin/GhcMod.hs

```diff
@@ -607,38 +608,102 @@ symbolProvider uri = pluginGetFile "ghc-mod symbolProvider: " uri $
+        goTyClD (L l (SynDecl { tcdLName = n })) = pure (Decl LSP.SkClass n [] l)
+        goTyClD (L l (DataDecl { tcdLName = n, tcdDataDefn = HsDataDefn { dd_cons = cons } })) =
           pure (Decl LSP.SkClass n (concatMap processCon cons) l)
-        go (L l (TyClD ClassDecl { tcdLName = n, tcdSigs = sigs, tcdATs = fams })) =
+        goTyClD (L l (ClassDecl { tcdLName = n, tcdSigs = sigs, tcdATs = fams })) =
           pure (Decl LSP.SkInterface n children l)
           where children = famDecls ++ sigDecls
+#if __GLASGOW_HASKELL__ >= 806
+                famDecls = concatMap (go . fmap (TyClD NoExt . FamDecl NoExt)) fams
+#else
                 famDecls = concatMap (go . fmap (TyClD . FamDecl)) fams
+#endif
                 sigDecls = concatMap processSig sigs
+#if __GLASGOW_HASKELL__ >= 806
+        goTyClD (L _ (FamDecl _ (XFamilyDecl _)))        = error "goTyClD"
+        goTyClD (L _ (DataDecl _ _ _ _ (XHsDataDefn _))) = error "goTyClD"
+        goTyClD (L _ (XTyClDecl _))                      = error "goTyClD"
+#endif
+
+        -- -----------------------------
 
-        go (L l (ValD FunBind { fun_id = ln, fun_matches = MG { mg_alts = llms } })) =
+        goValD :: LHsBind GM.GhcPs -> [Decl]
+        goValD (L l (FunBind { fun_id = ln, fun_matches = MG { mg_alts = llms } })) =
           pure (Decl LSP.SkFunction ln wheres l)
           where
             wheres = concatMap (gomatch . unLoc) (unLoc llms)
             gomatch Match { m_grhss = GRHSs { grhssLocalBinds = lbs } } = golbs (unLoc lbs)
+#if __GLASGOW_HASKELL__ >= 806
+            gomatch (Match _ _ _ (XGRHSs _)) = error "gomatch"
+            gomatch (XMatch _)               = error "gomatch"
+
+            golbs (HsValBinds _ (ValBinds _ lhsbs _)) = concatMap (go . fmap (ValD NoExt)) lhsbs
+#else
             golbs (HsValBinds (ValBindsIn lhsbs _ )) = concatMap (go . fmap ValD) lhsbs
+#endif
             golbs _ = []
 
-        go (L l (ValD PatBind { pat_lhs = p })) =
+        goValD (L l (PatBind { pat_lhs = p })) =
           map (\n -> Decl LSP.SkVariable n [] l) $ hsNamessRdr p
-        go (L l (ForD ForeignImport { fd_name = n })) = pure (Decl LSP.SkFunction n [] l)
-        go _ = []
+
+#if __GLASGOW_HASKELL__ >= 806
+        goValD (L _ (FunBind _ _ (XMatchGroup _) _ _)) = error "goValD"
+        goValD (L _ (VarBind _ _ _ _))                 = error "goValD"
+        goValD (L _ (AbsBinds _ _ _ _ _ _ _))          = error "goValD"
+        goValD (L _ (PatSynBind _ _))                  = error "goValD"
+        goValD (L _ (XHsBindsLR _))                    = error "goValD"
+#elif __GLASGOW_HASKELL__ >= 804
+        goValD (L _ (VarBind _ _ _))        = error "goValD"
+        goValD (L _ (AbsBinds _ _ _ _ _ _)) = error "goValD"
+        goValD (L _ (PatSynBind _))         = error "goValD"
+#else
+        goValD (L _ (VarBind _ _ _))           = error "goValD"
```

Comment:
Should these errors return `[]`?

Reply:
These are things that should never occur in the AST for `GhcPs`.  So  erroring out lets us know something is wrong, and should show up at dev time.

If anyone feels strongly about it ,I can make it `[]`, but that tells the next person who works on the code that these might be legitimate values.

Reply:
Ok that makes sense that they shouldn’t exist in the parse stage, I would rather have it crash in that case!